google/gax/__init__.py

```diff
@@ -588,10 +590,10 @@ def result(self, timeout=None):
 
     def exception(self, timeout=None):
         """Similar to result(), except returns the exception if any."""
-        if self._poll(timeout).HasField('error'):
-            return self._operation.error
+        if not self._poll(timeout).HasField('error'):
```

Comment:
@lukesneeringer I don't have feelings on prefering one format over the other *in general*, but I prefer to

- not make changes in a PR to existing code unless there's a justification (preferably stated in the commit message), and keep the patterns in any new code consistent with existing code
- put stylistic changes and functional changes in a separate PRs to make isolating regressions easier

That's more where this comment was coming from.